compiler/noirc_frontend/src/hir/comptime/hir_to_ast.rs

```diff
@@ -0,0 +1,345 @@
+                let element = Box::new(element.to_ast());
+                UnresolvedTypeData::FormatString(length, element)
+            }
+            Type::Unit => UnresolvedTypeData::Unit,
+            Type::Tuple(fields) => {
+                let fields = vecmap(fields, |field| field.to_ast());
+                UnresolvedTypeData::Tuple(fields)
+            }
+            Type::Struct(def, generics) => {
+                let struct_def = def.borrow();
+                let generics = vecmap(generics, |generic| generic.to_ast());
+                let name = Path::from_ident(struct_def.name.clone());
+                UnresolvedTypeData::Named(name, generics, false)
+            }
+            Type::Alias(type_def, generics) => {
+                // Keep the alias name instead of expanding this in case the
+                // alias' definition was changed
+                let type_def = type_def.borrow();
+                let generics = vecmap(generics, |generic| generic.to_ast());
+                let name = Path::from_ident(type_def.name.clone());
+                UnresolvedTypeData::Named(name, generics, false)
+            }
+            Type::TypeVariable(_, _) => todo!("Convert Type::TypeVariable Hir -> Ast"),
+            Type::TraitAsType(_, name, generics) => {
+                let generics = vecmap(generics, |generic| generic.to_ast());
+                let name = Path::from_single(name.as_ref().clone(), Span::default());
+                UnresolvedTypeData::TraitAsType(name, generics)
+            }
+            Type::NamedGeneric(_, name) => {
+                let name = Path::from_single(name.as_ref().clone(), Span::default());
+                UnresolvedTypeData::TraitAsType(name, Vec::new())
+            }
+            Type::Function(args, ret, env) => {
+                let args = vecmap(args, |arg| arg.to_ast());
+                let ret = Box::new(ret.to_ast());
+                let env = Box::new(env.to_ast());
+                UnresolvedTypeData::Function(args, ret, env)
+            }
+            Type::MutableReference(element) => {
+                let element = Box::new(element.to_ast());
+                UnresolvedTypeData::MutableReference(element)
+            }
+            // Type::Forall is only for generic functions which don't store a type
+            // in their Ast so they don't need to call to_ast for their Forall type.
+            // Since there is no UnresolvedTypeData equivalent for Type::Forall, we use
+            // this to ignore this case since it shouldn't be needed anyway.
+            Type::Forall(_, typ) => return typ.to_ast(),
+            Type::Constant(_) => panic!("Type::Constant where a type was expected: {self:?}"),
+            Type::Code => UnresolvedTypeData::Code,
+            Type::Error => UnresolvedTypeData::Error,
+        };
+
+        UnresolvedType { typ, span: None }
+    }
+
+    fn to_type_expression(&self) -> UnresolvedTypeExpression {
+        let span = Span::default();
```

Comment:
would it make sense to use `Option<Span>` in `UnresolvedTypeExpression` to distinguish these cases?

Reply:
The main problem is that we don't have spans for Types so we'd still have to choose either `None` or `Some(Span::default())` for everything. In the future we'll need to fix this I think but there isn't a satisfactory solution now.